src/attributes.jl

```diff
@@ -188,45 +193,82 @@ function h5readattr(filename, name::AbstractString)
 end
 
 
-struct Attributes
-    parent::Union{File,Object}
+struct AttributeDict <: AbstractDict{String,Any}
+    parent::Object
 end
 
 """
-    attributes(object::Union{File,Object})
+    attrs(object::Union{File,Group,Dataset,Datatype})
+
+The attributes dictionary of `object`. Returns an `AttributeDict`, a `Dict`-like
+object for accessing the attributes of `object`.
 
-The attributes of a file or object: this returns an `Attributes` object, which
-is `Dict`-like object for accessing the attributes of `object`: `getindex` will
-return an [`Attribute`](@ref) object, and `setindex!` will call [`write_attribute`](@ref).
+```julia
+attrs(object)["name"] = value  # create/overwrite an attribute
+attr = attrs(object)["name"]   # read an attribute
+delete!(attrs(object), "name") # delete an attribute
+keys(attrs(object))            # list the attribute names
+```
 """
-attributes(p::Union{File,Object}) = Attributes(p)
+function attrs(parent::Object)
+    return AttributeDict(parent)
+end
+attrs(file::File) = attrs(open_group(file, "."))
 
-Base.isvalid(obj::Attributes) = isvalid(obj.parent)
+Base.haskey(attrdict::AttributeDict, path::AbstractString) = API.h5a_exists(checkvalid(attrdict.parent), path)
+Base.length(attrdict::AttributeDict) = Int(object_info(attrdict.parent).num_attrs)
 
-function Base.getindex(x::Attributes, name::AbstractString)
+function Base.getindex(x::AttributeDict, name::AbstractString)
     haskey(x, name) || throw(KeyError(name))
-    open_attribute(x.parent, name)
```

Comment:
We can add the AttributeDict methods without removing the older interface. This still needs to work even when deprecated.